spec/features/05_user_updates_account_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+feature "user updates their account" do
+  scenario "user wants to update their account username" do
+    user = FactoryGirl.create(:user)
+    sign_up_as(user)
+    click_on "Profile"
+    click_on "Edit Profile"
+    fill_in "Username", with: "abcd"
+    fill_in "Current password", with: user.password
+    click_on "Update"
+    click_on "Profile"
+
+    expect(page).to have_content('Welcome, abcd!')
+  end
+
+  scenario "user wants to update their account password" do
+    user = FactoryGirl.create(:user)
+    sign_up_as(user)
+    click_on "Profile"
+    click_on "Edit Profile"
+    fill_in "Password", with: "abcd1234"
+    fill_in "Password confirmation", with: "abcd1234"
+    fill_in "Current password", with: user.password
+    click_on "Update"
+    click_on "Profile"
+
+    expect(page).to have_content("Welcome, #{user.username}1!")
+  end
```

Comment:
This doesn't really test if the password has changed. You should sign out and sign in with the new password.